src/troute-network/troute/AbstractNetwork.py

```diff
@@ -654,3 +658,201 @@ def _try_parsing_date(text):
+
+            # the number of files required for the simulation
+            nfiles = int(np.ceil(nts / qts_subdivisions))
+            
+            # list of forcing file datetimes
+            #datetime_list = [t0 + dt_qlat_timedelta * (n + 1) for n in
+            #                 range(nfiles)]
+            # ** Correction ** Because qlat file at time t is constantly applied throughout [t, t+1],
+            #               ** n + 1 should be replaced by n
+            datetime_list = [self.t0 + dt_qlat_timedelta * (n) for n in
+                            range(nfiles)]        
+            datetime_list_str = [datetime.strftime(d, '%Y%m%d%H%M') for d in
+                                datetime_list]
+
+            # list of forcing files
+            forcing_filename_list = [d_str + forcing_glob_filter[1:] for d_str in
+                                    datetime_list_str]
+            
+            # check that all forcing files exist
+            for f in forcing_filename_list:
+                try:
+                    J = pathlib.Path(qlat_input_folder.joinpath(f))     
+                    assert J.is_file() == True
+                except AssertionError:
+                    raise AssertionError("Aborting simulation because forcing file", J, "cannot be not found.") from None
+                    
+            # build run sets list
+            run_sets = []
+            k = 0
+            j = 0
+            nts_accum = 0
+            nts_last = 0
+            while k < len(forcing_filename_list):
+                run_sets.append({})
+
+                if k + max_loop_size < len(forcing_filename_list):
+                    run_sets[j]['qlat_files'] = forcing_filename_list[k:k
+                        + max_loop_size]
+                else:
+                    run_sets[j]['qlat_files'] = forcing_filename_list[k:]
+
+                nts_accum += len(run_sets[j]['qlat_files']) * qts_subdivisions
+                if nts_accum <= nts:
+                    run_sets[j]['nts'] = int(len(run_sets[j]['qlat_files'])
+                                            * qts_subdivisions)
+                else:
+                    run_sets[j]['nts'] = int(nts - nts_last)
+
+                final_qlat = qlat_input_folder.joinpath(run_sets[j]['qlat_files'][-1]) 
+                if forcing_glob_filter=="*.CHRTOUT_DOMAIN1":           
+                    final_timestamp_str = nhd_io.get_param_str(final_qlat,'model_output_valid_time')
+                else:
+                    df = read_file(final_qlat)
+                    final_timestamp_str = pd.to_datetime(df.columns[1]).strftime("%Y-%m-%d_%H:%M:%S")           
+                
+                run_sets[j]['final_timestamp'] = \
+                    datetime.strptime(final_timestamp_str, '%Y-%m-%d_%H:%M:%S')
+
+                nts_last = nts_accum
+                k += max_loop_size
+                j += 1
+
+        return run_sets
+
+def nex_files_to_binary(nexus_files, binary_folder):
```

Comment:
Also, this is HYFeatures specific, and as such this function should exist in that sublcass, not the abstract class.